compiler/rustc_passes/src/reachable.rs

```diff
@@ -1,14 +1,26 @@
-//! Finds local items that are externally reachable, which means that other crates need access to
-//! their compiled machine code or their MIR.
+//! Finds local items that are "reachable", which means that other crates need access to their
+//! compiled machine code or their *runtime* MIR. (Compile-time MIR is always encoded anyway, so we
+//! don't worry about that here.)
 //!
-//! An item is "externally reachable" if it is relevant for other crates. This obviously includes
-//! all public items. However, some of these items cannot be compiled to machine code (because they
-//! are generic), and for some the machine code is not sufficient (because we want to cross-crate
-//! inline them). These items "need cross-crate MIR". When a reachable function `f` needs
-//! cross-crate MIR, then all the functions it calls also become reachable, as they will be
-//! necessary to use the MIR of `f` from another crate. Furthermore, an item can become "externally
-//! reachable" by having a `const`/`const fn` return a pointer to that item, so we also need to
-//! recurse into reachable `const`/`const fn`.
+//! An item is "reachable" if codegen that happens in downstream crates can end up referencing this
+//! item. This obviously includes all public items. However, some of these items cannot be compiled
+//! to machine code (because they are generic), and for some the machine code is not sufficient
```

Comment:
```suggestion
//! item. This obviously includes all public items. However, some of these items cannot be lowered
//! (because they are generic), and for some the machine code is not sufficient
```
:shrug: I dislike mentioning "machine code" because a codegen backend can do things like emit JavaScript, but I don't have a particularly compelling alternative in mind.

Reply:
Makes sense, I like the alternative